src/Features/Core/Completion/Providers/AbstractSymbolCompletionProvider.cs

```diff
@@ -19,6 +19,14 @@ namespace Microsoft.CodeAnalysis.Completion.Providers
 {
     internal abstract partial class AbstractSymbolCompletionProvider : AbstractCompletionProvider
     {
+        // PERF: Many CompletionProviders derive AbstractSymbolCompletionProvider and therefore
+        // compute identical contexts. This actually shows up on the 2-core typing test.
+        // Cache the most recent document/position/computed SyntaxContext to reduce repeat computation.
+        private static Document cachedDocument;
```

Comment:
Roslyn now uses the .NET Foundation coding style

> https://github.com/dotnet/corefx/wiki/Contributing#c-coding-style

This specifies private fields begin with an `_`.  

This is referenced now from our [Contributing Code](https://github.com/dotnet/roslyn/wiki/Contributing-Code) wiki page.